scipy/optimize/_zeros_py.py

```diff
@@ -1393,3 +1393,328 @@ def toms748(f, a, b, args=(), k=1,
     x, function_calls, iterations, flag = result
     return _results_select(full_output, (x, function_calls, iterations, flag),
                            "toms748")
+
+
+def muller(f, x0, x1, x2, args=(), tol=1.48e-8, maxiter=50,
```

Comment:
Again for simplicity let's start with this only exposed through `root_scalar` then in the future we can decide whether to expose through newton or it's own method

Reply:
Would you like me to change this to `_muller()`, or to stuff it into another function?

Reply:
`_muller()` sounds good